Content.Client/VendingMachines/VendingMachineMenu.cs

```diff
@@ -42,17 +42,25 @@ public void Populate(List<VendingMachineInventoryEntry> inventory)
         {
             _items.Clear();
             _cachedInventory = inventory;
+            var longestEntry = "";
             foreach (VendingMachineInventoryEntry entry in inventory)
             {
                 var itemName = _prototypeManager.Index<EntityPrototype>(entry.ID).Name;
+                if (itemName.Length > longestEntry.Length)
+                {
+                    longestEntry = itemName;
+                }
 
                 Texture? icon = null;
                 if(_prototypeManager.TryIndex(entry.ID, out EntityPrototype? prototype))
                 {
                     icon = SpriteComponent.GetPrototypeIcon(prototype, _resourceCache).Default;
                 }
+
                 _items.AddItem($"{itemName} ({entry.Amount} left)", icon);
             }
+
+            MinSize = SetSize = ((longestEntry.Length + 8) * 12, _items.Count * 40 + 50);
```

Comment:
please tell me there is a better way to do this because I'm not familiar with UI in this codebase

Reply:
_n-no_